File: API/Domain/ExerciseDiaries/ExerciseDiary.cs
using Domain.Exercises;

namespace Domain.ExerciseDiaries;

public record ExerciseDiary(
    Guid Id,
    Guid UserId,
    DateOnly Date,
    ExerciseDiaryRecord Current,
    ExerciseDiaryRecord Target);

public record ExerciseDiaryRecord(
    int Steps,
    ICollection<CompletedExercise>? CompletedExercises,
    ExerciseDiaryRecordInfo Info);
    
public record CompletedExercise(
    Exercise Exercise,
    int? Repeats,
    int? TimeInSeconds);
    
public record ExerciseDiaryRecordInfo(
    float TotalKcalBurnt);

File: client/src/app/features/tabs/features/food-diary-tab/components/exercise-diary/types/ExerciseDiary.ts
import {ExerciseDiaryRecord} from "./ExerciseDiaryRecord";

export type ExerciseDiary = {
  "id": string,
  "userId": string,
  "date": string,
  "current": ExerciseDiaryRecord,
  "target": ExerciseDiaryRecord
}
